File: 🟢 Easy/2085 - Count Common Words With One Occurrence/_2085CountCommonWordsWithOneOccurrence.py
class Solution:
    def countWords(self, words1: List[str], words2: List[str]) -> int:
        map1 = {}
        map2 = {}
        count = 0

        for str in words1:
            if str in map1:
                map1[str] += 1
            else:
                map1[str] = 1

        for str in words2:
            if str in map2:
                map2[str] += 1
            else:
                map2[str] = 1

        for str in words1:
            if map1[str] == 1:
                if str in map2:
                    if map2[str] == 1:
                        count += 1

        return count

File: 🟢 Easy/2085 - Count Common Words With One Occurrence/_2085CountCommonWordsWithOneOccurrence.cpp
class Solution {
public:
    int countWords(vector<string>& words1, vector<string>& words2) {
        unordered_map<string, int> map1;
        unordered_map<string, int> map2;
        int count = 0;

        for (auto& str : words1) {
            if (map1.find(str) != map1.end()) {
                map1[str]++;
            } else {
                map1[str] = 1;
            }
        }

        for (auto& str : words2) {
            if (map2.find(str) != map2.end()) {
                map2[str]++;
            } else {
                map2[str] = 1;
            }
        }

        for (auto& str : words1) {
            if (map1[str] == 1 && map2.find(str) != map2.end()) {
                if (map2[str] == 1) {
                    count++;
                }
            }
        }

        return count;
    }
};